go/tasks/pluginmachinery/google/gke_token_source.go

```diff
@@ -0,0 +1,212 @@
+package google
+
+import (
+	"context"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/flyteorg/flytestdlib/atomic"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/golang/groupcache/singleflight"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	v1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	defaultGracePeriod             = 300
+	gcpServiceAccountAnnotationKey = "iam.gke.io/gcp-service-account"
+	workflowIdentityDocURL         = "https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity"
+)
+
+type gkeTokenSource struct {
+	kubeClient             kubernetes.Interface
+	tokens                 *sync.Map
```

Comment:
why not use one of the ttl caches we have an implementation for already?

Reply:
can you please point out to TTL cache? I've only found auto refresh cache

Reply:
ohh here is an example using one - https://github.com/flyteorg/flytepropeller/blob/master/pkg/controller/nodes/task/barrier.go#L28

We just use the k8s apimachinery cache

Reply:
Thanks! This is what I was looking for